modules/libjpeg_turbo/2.1.4/patches/add_build_file.patch

```diff
@@ -0,0 +1,792 @@
++          "%define EXT_RGBX_RED 0\n" +
++          "%define EXT_RGBX_GREEN 1\n" +
++          "%define EXT_RGBX_BLUE 2\n" +
++          "%define EXT_RGBX_PIXELSIZE 4\n" +
++          "%define EXT_BGR_RED 2\n" +
++          "%define EXT_BGR_GREEN 1\n" +
++          "%define EXT_BGR_BLUE 0\n" +
++          "%define EXT_BGR_PIXELSIZE 3\n" +
++          "%define EXT_BGRX_RED 2\n" +
++          "%define EXT_BGRX_GREEN 1\n" +
++          "%define EXT_BGRX_BLUE 0\n" +
++          "%define EXT_BGRX_PIXELSIZE 4\n" +
++          "%define EXT_XBGR_RED 3\n" +
++          "%define EXT_XBGR_GREEN 2\n" +
++          "%define EXT_XBGR_BLUE 1\n" +
++          "%define EXT_XBGR_PIXELSIZE 4\n" +
++          "%define EXT_XRGB_RED 1\n" +
++          "%define EXT_XRGB_GREEN 2\n" +
++          "%define EXT_XRGB_BLUE 3\n" +
++          "%define EXT_XRGB_PIXELSIZE 4\n" +
++          "%define RGBX_FILLER_0XFF 1\n" +
++          "%define JSAMPLE byte ; unsigned char\n" +
++          "%define SIZEOF_JSAMPLE SIZEOF_BYTE ; sizeof(JSAMPLE)\n" +
++          "%define CENTERJSAMPLE 128\n" +
++          "%define JCOEF word ; short\n" +
++          "%define SIZEOF_JCOEF SIZEOF_WORD ; sizeof(JCOEF)\n" +
++          "%define JDIMENSION dword ; unsigned int\n" +
++          "%define SIZEOF_JDIMENSION SIZEOF_DWORD ; sizeof(JDIMENSION)\n" +
++          "%define JSAMPROW POINTER ; JSAMPLE * (jpeglib.h)\n" +
++          "%define JSAMPARRAY POINTER ; JSAMPROW * (jpeglib.h)\n" +
++          "%define JSAMPIMAGE POINTER ; JSAMPARRAY * (jpeglib.h)\n" +
++          "%define JCOEFPTR POINTER ; JCOEF * (jpeglib.h)\n" +
++          "%define SIZEOF_JSAMPROW SIZEOF_POINTER ; sizeof(JSAMPROW)\n" +
++          "%define SIZEOF_JSAMPARRAY SIZEOF_POINTER ; sizeof(JSAMPARRAY)\n" +
++          "%define SIZEOF_JSAMPIMAGE SIZEOF_POINTER ; sizeof(JSAMPIMAGE)\n" +
++          "%define SIZEOF_JCOEFPTR SIZEOF_POINTER ; sizeof(JCOEFPTR)\n" +
++          "%define DCTELEM word ; short\n" +
++          "%define SIZEOF_DCTELEM SIZEOF_WORD ; sizeof(DCTELEM)\n" +
++          "%define float FP32 ; float\n" +
++          "%define SIZEOF_FAST_FLOAT SIZEOF_FP32 ; sizeof(float)\n" +
++          "%define ISLOW_MULT_TYPE word ; must be short\n" +
++          "%define SIZEOF_ISLOW_MULT_TYPE SIZEOF_WORD ; sizeof(ISLOW_MULT_TYPE)\n" +
++          "%define IFAST_MULT_TYPE word ; must be short\n" +
++          "%define SIZEOF_IFAST_MULT_TYPE SIZEOF_WORD ; sizeof(IFAST_MULT_TYPE)\n" +
++          "%define IFAST_SCALE_BITS 2 ; fractional bits in scale factors\n" +
++          "%define FLOAT_MULT_TYPE FP32 ; must be float\n" +
++          "%define SIZEOF_FLOAT_MULT_TYPE SIZEOF_FP32 ; sizeof(FLOAT_MULT_TYPE)\n" +
++          "%define JSIMD_NONE 0x00\n" +
++          "%define JSIMD_MMX 0x01\n" +
++          "%define JSIMD_3DNOW 0x02\n" +
++          "%define JSIMD_SSE 0x04\n" +
++          "%define JSIMD_SSE2 0x08\n" +
++          "EOF",
++)
++
++config_setting(
++    name = "k8",
++    values = {"cpu": "k8"},
```

Comment:
These kinds of settings will no longer work with Bazel 7, which flipped `--incompatible_enable_cc_toolchain_resolution`. Do you know whether upstream ready has a plan to migrate?

Cc @meteorcloudy as this comes from Tensorflow

Reply:
@vertexwahn Let me know if you need support in getting this ready. We don't necessarily have to fix these issues, but if they break users on Bazel 7, that may not be ideal.

Reply:
Sorry for the silence here. I agree this isn't very nice, it'll be great if it can be updated to use platform. But I'm fine with checking in as it is for the initial version, we can polish the BUILD file in later BCR versions.